src/components/common/warning-banner/__tests__/warning-banner.spec.ts

```diff
@@ -0,0 +1,38 @@
+import { TestBed, async, ComponentFixture } from '@angular/core/testing';
+import { IonicModule, Config } from 'ionic-angular';
+import { ConfigMock } from 'ionic-mocks';
+import { By } from '@angular/platform-browser';
+import { WarningBannerComponent } from '../warning-banner';
+
+describe('WarningBanner', () => {
+  let fixture: ComponentFixture<WarningBannerComponent>;
+  let component: WarningBannerComponent;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [WarningBannerComponent],
+      imports: [
+        IonicModule,
+      ],
+      providers: [
+        { provide: Config, useFactory: () => ConfigMock.instance() },
+      ],
+    })
+      .compileComponents()
+      .then(() => {
+        fixture = TestBed.createComponent(WarningBannerComponent);
+        component = fixture.componentInstance;
+      });
+  }));
+
+  describe('DOM', () => {
+    it('should display the warning message', () => {
+      const warningText = 'This is the warning text';
+      component.warningText = warningText;
+
+      fixture.detectChanges();
+      const rendered = fixture.debugElement.query(By.css('span')).nativeElement.innerHTML;
```

Comment:
Can this not use an id to select - if extra spans are added to the html at a later date it could potentially cause this test to fail so it's a bit brittle